utils/telegram.py

```diff
@@ -17,20 +18,21 @@ class TelegramHandler:
 
     SLICE_LENGTH = 4096
     
-    def __init__(self, api_token: str, conversations = list):
+    def __init__(self, api_token: str, conversations: dict):
         self.updater = Updater(api_token, use_context=True)
         self.conversations = conversations
         self.updater.dispatcher.add_handler(CommandHandler('start', Handlers.start))
             
     def poll(self):
         self.updater.start_polling()
 
-    def broadcast(self, message):
+    def broadcast(self, message, filter_str=None):
         bot = self.updater.bot
         for chunk in self.chunks(message):
-            for conversation in self.conversations:
-                bot.send_chat_action(chat_id=conversation, action=ChatAction.TYPING)
-                bot.send_message(chat_id=conversation, text=chunk)
+            for conversation,filter_regex in self.conversations.items():
+                if filter_str and re.match(filter_regex, filter_str):
```

Comment:
if `filter_str` is empty, we should allow the message to go to __every__ chat, instead of __no__ chat. That allows the function to still be generic for other broadcasts. Should we need a function to send to only filtered chats, we'll create another function (and not call it `broadcast`).